radix.py

```diff
@@ -0,0 +1,272 @@
+    ).timeit(10) / 10
+
+    print
+    print "Time for reverse case of list of first 3000 integers"
+    print Timer(
+        'radix_integer(reverse_case(3000))',
+        'from __main__ import radix_integer, reverse_case'
+    ).timeit(10) / 10
+
+    print
+    print "Time for all ones case of list of first 3000 integers"
+    print Timer(
+        'radix_integer(ones(3000))',
+        'from __main__ import radix_integer, ones'
+    ).timeit(10) / 10
+
+    print
+    print "Time for shuffled case of list of first 3000 integers"
+    print Timer(
+        'radix_integer(shuffled_case(3000))',
+        'from __main__ import radix_integer, shuffled_case'
+    ).timeit(10) / 10
+
+    print
+    print "Time for best case of list of first 4000 integers"
+    print Timer(
+        'radix_integer(best_case(4000))',
+        'from __main__ import radix_integer, best_case'
+    ).timeit(10) / 10
+
+    print
+    print "Time for reverse case of list of first 4000 integers"
+    print Timer(
+        'radix_integer(reverse_case(4000))',
+        'from __main__ import radix_integer, reverse_case'
+    ).timeit(10) / 10
+
+    print
+    print "Time for all ones case of list of first 4000 integers"
+    print Timer(
+        'radix_integer(ones(4000))',
+        'from __main__ import radix_integer, ones'
+    ).timeit(10) / 10
+
+    print
+    print "Time for shuffled case of list of first 4000 integers"
+    print Timer(
+        'radix_integer(shuffled_case(4000))',
+        'from __main__ import radix_integer, shuffled_case'
+    ).timeit(10) / 10
+
+    print
+    print "Time for best case of list of first 5000 integers"
+    print Timer(
+        'radix_integer(best_case(5000))',
+        'from __main__ import radix_integer, best_case'
+    ).timeit(10) / 10
+
+    print
+    print "Time for reverse case of list of first 5000 integers"
+    print Timer(
+        'radix_integer(reverse_case(5000))',
+        'from __main__ import radix_integer, reverse_case'
+    ).timeit(10) / 10
+
+    print
+    print "Time for all ones case of list of first 5000 integers"
+    print Timer(
+        'radix_integer(ones(5000))',
+        'from __main__ import radix_integer, ones'
+    ).timeit(10) / 10
+
+    print
+    print "Time for shuffled case of list of first 5000 integers"
+    print Timer(
+        'radix_integer(shuffled_case(5000))',
+        'from __main__ import radix_integer, shuffled_case'
+    ).timeit(10) / 10
+
+    print "These test cases show similar behavior for sorted, reverse sorted,"
+    print "and shuffled data sets. However, the case for all equal values"
+    print "performs exceptionally well. Possibly because so many of the"
+    print "buckets collapse out early, reducing the number of iterations."
+    print "All cases show several times faster behavior than the alternatives,"
+    print "with a large advantage for similar data within a list. The time"
+    print "complexity approaches O(n) for cases tested."
```

Comment:
so what is the time complexity of the radix sort dependent on?  Can you identify the controlling factor?